invenio_rdm_records/services/components/pids.py

```diff
@@ -26,13 +26,19 @@ def create(self, identity, data=None, record=None, errors=None):
 
         It validates and add the pids to the draft.
         """
-        pids = data.get("pids", {})
+        if "pids" in data:
+            pids = data.get("pids", {})
+        else:
+            pids = record.pids or {}
```

Comment:
Can you test the use case (currently via the UI):

- Fill in an external pid
- Save
- Change your answer from "yes" to "no"
- Save again. Payload sent is skipping the `pids` key entirely and the UI expects the backend to reserve one when the record is published!